File: workflow/java/src/main/java/io/dapr/quickstarts/workflows/activities/NotifyActivity.java
package io.dapr.quickstarts.workflows.activities;

import io.dapr.quickstarts.workflows.models.Notification;
import io.dapr.workflows.runtime.WorkflowActivity;
import io.dapr.workflows.runtime.WorkflowActivityContext;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class NotifyActivity implements WorkflowActivity {

  private static final Logger logger = LoggerFactory.getLogger(NotifyActivity.class);

  @Override
  public Object run(WorkflowActivityContext ctx) {
    Notification notification = ctx.getInput(Notification.class);
    logger.info(notification.getMessage());
    return null;
  }
}

File: workflow/csharp/Activities/NotifyActivity.cs
namespace WorkflowApp.Activities
{
    using System.Threading.Tasks;
    using Dapr.Workflow;
    using Microsoft.Extensions.Logging;
    using WorkflowApp.Models;

    public class NotifyActivity : WorkflowActivity<Notification, object>
    {
        readonly ILogger logger;

        public NotifyActivity(ILoggerFactory loggerFactory)
        {
            this.logger = loggerFactory.CreateLogger<NotifyActivity>();
        }

        public override Task<object> RunAsync(WorkflowActivityContext context, Notification notification)
        {
            this.logger.LogInformation(notification.Message);
            return Task.FromResult<object>(null!);
        }
    }
}

